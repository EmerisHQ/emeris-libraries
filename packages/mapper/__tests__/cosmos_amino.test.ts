import { coin, coins, makeSignDoc, Secp256k1HdWallet } from "@cosmjs/amino";
import CosmosAminoMessageMapper from "../src/cosmos_amino";

describe('Cosmos Mapps', () => {
    const mnemonic = "surround miss nominee dream gap cross assault thank captain prosper drop duty group candy wealth weather scale put";
    const chainId = "emeris-chain"
    const fee = {
        amount: coins(2000, "ucosm"),
        gas: "180000", // 180k
    };
    let firstAccount, mapper, wallet

    beforeAll(async () => {
        wallet = await Secp256k1HdWallet.fromMnemonic(mnemonic);
        [firstAccount] = await wallet.getAccounts();

        mapper = new CosmosAminoMessageMapper(chainId)
    })

    it('transfer', async () => {
        const msgs = mapper.transfer({
            amount: coins(2000, "ucosm"),
            to_address: firstAccount.address,
            chain_name: chainId
        }, firstAccount.address)

        const signDoc = makeSignDoc(msgs, fee, chainId, "memo", 0, 0);
        const { signed, signature } = await wallet.signAmino(firstAccount.address as string, signDoc);

        expect({ signed, signature }).toMatchSnapshot();
    });

    it('ibcTransfer', async () => {
        jest
            .useFakeTimers('modern')
            .setSystemTime(new Date('2020-01-01').getTime());

        const msgs = mapper.ibcTransfer({
            from_chain: chainId,
            to_chain: "moon",
            through: "abcdchannel",
            to_address: firstAccount.address,
            amount: coins(2000, "ucosm"),
        }, firstAccount.address)

        const signDoc = makeSignDoc(msgs, fee, chainId, "memo", 0, 0);
        const { signed, signature } = await wallet.signAmino(firstAccount.address, signDoc);

        expect({ signed, signature }).toMatchSnapshot();
    });

    it('swap', async () => {
        const msgs = mapper.swap({
            pool: {
                id: 1,
                type_id: 1,
                reserve_coin_denoms: ["sayajin", "supersajayin"]
            },
            from: coin(100, "sayajin"),
            to: coin(10, "supersajayin"),
            orderPrice: "10500000000000000000",
            offerFee: {
                "amount": "5",
                "denom": "sayajin",
            }
        }, firstAccount.address)

        const signDoc = makeSignDoc(msgs, fee, chainId, "memo", 0, 0);
        const { signed, signature } = await wallet.signAmino(firstAccount.address, signDoc);

        expect({ signed, signature }).toMatchSnapshot();
    });

    it('addliquidity', async () => {
        const msgs = mapper.addliquidity({
            pool: {
                id: 1,
                type_id: 1,
                reserve_coin_denoms: ["sayajin", "supersajayin"]
            },
            coinA: coin(100, "sayajin"),
            coinB: coin(10, "supersajayin"),
        }, firstAccount.address)

        const signDoc = makeSignDoc(msgs, fee, chainId, "memo", 0, 0);
        const { signed, signature } = await wallet.signAmino(firstAccount.address, signDoc);

        expect({ signed, signature }).toMatchSnapshot();
    });

    it('withdrawliquidity', async () => {
        const msgs = mapper.withdrawliquidity({
            pool: {
                id: 1,
                type_id: 1
            },
            poolCoin: coin(100, "sayajin"),
        }, firstAccount.address)

        const signDoc = makeSignDoc(msgs, fee, chainId, "memo", 0, 0);
        const { signed, signature } = await wallet.signAmino(firstAccount.address, signDoc);

        expect({ signed, signature }).toMatchSnapshot();
    });

    it('createpool', async () => {
        const msgs = mapper.createpool({
            pool: {
                id: 1,
                type_id: 1
            },
            coinA: coin(100, "sayajin"),
            coinB: coin(10, "supersajayin"),
        }, firstAccount.address)

        const signDoc = makeSignDoc(msgs, fee, chainId, "memo", 0, 0);
        const { signed, signature } = await wallet.signAmino(firstAccount.address, signDoc);

        expect({ signed, signature }).toMatchSnapshot();
    });
})