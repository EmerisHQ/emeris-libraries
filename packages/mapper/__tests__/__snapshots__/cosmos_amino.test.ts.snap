// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cosmos Mapps addliquidity 1`] = `
Object {
  "signature": Object {
    "pub_key": Object {
      "type": "tendermint/PubKeySecp256k1",
      "value": "A4Fj1Y4k77Qaxuy496CHYB2rpfWXkM3LCnlyrU8eKbH7",
    },
    "signature": "2thzra3mWrp16gmZfdo2dt/zeHp7jYz9QeuTqnHiHw16l5kJZVwMm6zK16QgB0WWlJFBj8Z8Ry33I0Zy4UCxOw==",
  },
  "signed": Object {
    "account_number": "0",
    "chain_id": "emeris-chain",
    "fee": Object {
      "amount": Array [
        Object {
          "amount": "2000",
          "denom": "ucosm",
        },
      ],
      "gas": "180000",
    },
    "memo": "memo",
    "msgs": Array [
      Object {
        "type": "liquidity/MsgDepositWithinBatch",
        "value": Object {
          "deposit_coins": Array [
            Object {
              "amount": "100",
              "denom": "sayajin",
            },
            Object {
              "amount": "10",
              "denom": "supersajayin",
            },
          ],
          "depositor_address": "cosmos1lsagfzrm4gz28he4wunt63sts5xzmczw8pkek3",
          "pool_id": 1,
        },
      },
    ],
    "sequence": "0",
  },
}
`;

exports[`Cosmos Mapps createpool 1`] = `
Object {
  "signature": Object {
    "pub_key": Object {
      "type": "tendermint/PubKeySecp256k1",
      "value": "A4Fj1Y4k77Qaxuy496CHYB2rpfWXkM3LCnlyrU8eKbH7",
    },
    "signature": "ke4AADGJQiiFcNHc2exQV5Uf4//2fDM48WkshJWTJhp2+ZLXLKH0WpVVjVfkkk9sYUArQSv5LryTjOK9suWgkQ==",
  },
  "signed": Object {
    "account_number": "0",
    "chain_id": "emeris-chain",
    "fee": Object {
      "amount": Array [
        Object {
          "amount": "2000",
          "denom": "ucosm",
        },
      ],
      "gas": "180000",
    },
    "memo": "memo",
    "msgs": Array [
      Object {
        "type": "liquidity/MsgCreatePool",
        "value": Object {
          "deposit_coins": Array [
            Object {
              "amount": "100",
              "denom": "sayajin",
            },
            Object {
              "amount": "10",
              "denom": "supersajayin",
            },
          ],
          "pool_creator_address": "cosmos1lsagfzrm4gz28he4wunt63sts5xzmczw8pkek3",
          "pool_type_id": 1,
        },
      },
    ],
    "sequence": "0",
  },
}
`;

exports[`Cosmos Mapps ibcTransfer 1`] = `
Object {
  "signature": Object {
    "pub_key": Object {
      "type": "tendermint/PubKeySecp256k1",
      "value": "A4Fj1Y4k77Qaxuy496CHYB2rpfWXkM3LCnlyrU8eKbH7",
    },
    "signature": "GoucSewMkuOH8TenSawi4gzBac0MaNijwrG82RXudrhm/CL24eSf7NONrbAlYWbAXZskAo7gw0FJr0Jm+dhLUw==",
  },
  "signed": Object {
    "account_number": "0",
    "chain_id": "emeris-chain",
    "fee": Object {
      "amount": Array [
        Object {
          "amount": "2000",
          "denom": "ucosm",
        },
      ],
      "gas": "180000",
    },
    "memo": "memo",
    "msgs": Array [
      Object {
        "type": "cosmos-sdk/MsgTransfer",
        "value": Object {
          "receiver": "cosmos1lsagfzrm4gz28he4wunt63sts5xzmczw8pkek3",
          "sender": "cosmos1lsagfzrm4gz28he4wunt63sts5xzmczw8pkek3",
          "source_channel": "abcdchannel",
          "source_port": "transfer",
          "timeout_timestamp": Long {
            "high": 367368827,
            "low": -1829849088,
            "unsigned": false,
          },
          "token": Object {
            "0": Object {
              "amount": "2000",
              "denom": "ucosm",
            },
          },
        },
      },
    ],
    "sequence": "0",
  },
}
`;

exports[`Cosmos Mapps swap 1`] = `
Object {
  "signature": Object {
    "pub_key": Object {
      "type": "tendermint/PubKeySecp256k1",
      "value": "A4Fj1Y4k77Qaxuy496CHYB2rpfWXkM3LCnlyrU8eKbH7",
    },
    "signature": "1P+Yn1vbx2ounpqNPQ+T8eDvif94dNpfL0ekXJWoou9NLtS/ZrmAlZaIAoGvdexd87DelH0Kp3g77Uy0rNiNwA==",
  },
  "signed": Object {
    "account_number": "0",
    "chain_id": "emeris-chain",
    "fee": Object {
      "amount": Array [
        Object {
          "amount": "2000",
          "denom": "ucosm",
        },
      ],
      "gas": "180000",
    },
    "memo": "memo",
    "msgs": Array [
      Object {
        "type": "liquidity/MsgSwapWithinBatch",
        "value": Object {
          "demand_coin_denom": "supersajayin",
          "offer_coin": Object {
            "amount": "100",
            "denom": "sayajin",
          },
          "offer_coin_dee": Object {
            "amount": "5",
            "denom": "sayajin",
          },
          "order_price": "10500000000000000000",
          "pool_id": 1,
          "swap_requester_address": "cosmos1lsagfzrm4gz28he4wunt63sts5xzmczw8pkek3",
          "swap_type_id": 1,
        },
      },
    ],
    "sequence": "0",
  },
}
`;

exports[`Cosmos Mapps transfer 1`] = `
Object {
  "signature": Object {
    "pub_key": Object {
      "type": "tendermint/PubKeySecp256k1",
      "value": "A4Fj1Y4k77Qaxuy496CHYB2rpfWXkM3LCnlyrU8eKbH7",
    },
    "signature": "eL058wrn5N2EDEfS5RzInEehBNJspMMkOJiYzTW47RAfBkzD8Tbd7r26kkucI5DmNSd41RKHmeDfy93uRbkz6w==",
  },
  "signed": Object {
    "account_number": "0",
    "chain_id": "emeris-chain",
    "fee": Object {
      "amount": Array [
        Object {
          "amount": "2000",
          "denom": "ucosm",
        },
      ],
      "gas": "180000",
    },
    "memo": "memo",
    "msgs": Array [
      Object {
        "type": "cosmos-sdk/MsgSend",
        "value": Object {
          "amount": Array [
            Array [
              Object {
                "amount": "2000",
                "denom": "ucosm",
              },
            ],
          ],
          "from_address": "cosmos1lsagfzrm4gz28he4wunt63sts5xzmczw8pkek3",
          "to_address": "cosmos1lsagfzrm4gz28he4wunt63sts5xzmczw8pkek3",
        },
      },
    ],
    "sequence": "0",
  },
}
`;

exports[`Cosmos Mapps withdrawliquidity 1`] = `
Object {
  "signature": Object {
    "pub_key": Object {
      "type": "tendermint/PubKeySecp256k1",
      "value": "A4Fj1Y4k77Qaxuy496CHYB2rpfWXkM3LCnlyrU8eKbH7",
    },
    "signature": "b+fPhTi91MTPvLzXuoQcBwCfbQ9llo24S2GePlxq+kFTV0vMiXzOillrN95r2olnDBtZ7rnIA7Kjl1ocRfAISQ==",
  },
  "signed": Object {
    "account_number": "0",
    "chain_id": "emeris-chain",
    "fee": Object {
      "amount": Array [
        Object {
          "amount": "2000",
          "denom": "ucosm",
        },
      ],
      "gas": "180000",
    },
    "memo": "memo",
    "msgs": Array [
      Object {
        "type": "liquidity/MsgWithdrawWithinBatch",
        "value": Object {
          "pool_coin": Object {
            "amount": "100",
            "denom": "sayajin",
          },
          "pool_id": 1,
          "withdrawer_address": "cosmos1lsagfzrm4gz28he4wunt63sts5xzmczw8pkek3",
        },
      },
    ],
    "sequence": "0",
  },
}
`;
